QualificationChecker/Controllers/CandidatesController.cs

```diff
@@ -11,100 +13,63 @@ public partial class CandidatesController : ControllerBase
-                        var lookupKey = candidateAnswer.PositionId;
 
-                        if (orgAnswersDictionary.TryGetValue(lookupKey, out var OrgAnswers))
+                        // only run this if they pass the knockout questions
+                        if (myCandidate.Candidate.OrganizationId == myCandidate.Organization.OrgId && myCandidate.IsInterested() && myCandidate.MeetsAgeRequirement())
                         {
-                            // Loop through the orgAnswers (in case there are multiple with the same PositionId)
-                            bool answersMatch = false;
 
-
-                            foreach (var OrgAnswer in OrgAnswers)
+                            // Check to make sure postionId makes a dictionary 
+                            if (myCandidate.OrgAnswersDictionary != null)
                             {
-                                // make sure they have the same number first before checking answers
-                                if (OrgAnswers.Count != candidate.CandidateQuestions.Count)
-                                {
-                                    answersMatch = false;
-                                    break;
-                                }
-
-                                if (!MatchAnswers(candidateAnswer.Answer, OrgAnswer.Answer))
-                                {
-                                    // If an answer doesn't match or orgAnswer is null, set answersMatch to false
-                                    answersMatch = false;
-                                    break;
-                                }
-                                answersMatch = true;
+                                QualifiedCandidates.AddRange(myCandidate.CheckQuestions());
                             }
-
-                            if (!answersMatch)
-                            {
-                                // If no orgAnswer matches, set the flag to false and break the loop
-                                allAnswersMatch = false;
-                                break;
-                            }
-                        }
-                        else
-                        {
-                            // If no orgAnswer is found for this PositionId, set the flag to false and break the loop
-                            allAnswersMatch = false;
-                            break;
                         }
                     }
-
-                    if (allAnswersMatch)
+                    if (QualifiedCandidates.Count >= 1)
+                    { // Return the list of qualified candidates
+                        return Ok(QualifiedCandidates);
+                    }
+                    else
                     {
-                        // Add the candidate to the list of qualified candidates
-                        qualifiedCandidates.Add(candidate);
+                        return Ok("No Matching Canidates");
```

Comment:
Probably should not return OK - since validation failed.

Reply:
I was thinking that but you could still have no matching candidate and that be ok to message the front end no matching candidates

Reply:
Snagajob way would say return a different status code (412) ... but up to you.